dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -1032,13 +1061,44 @@ private String addOrUpdateDataset(IndexableDataset indexableDataset) {
                         filesToIndex.add(datafileSolrInputDocument);
                     }
                 }
+                
+                if (firstFileTermsOfUse == null) {
+                    firstFileTermsOfUse = fileMetadata.getTermsOfUse();
+                } else if (sameLicenseForAllFiles) {
+                    if (firstFileTermsOfUse.getTermsOfUseType() != fileMetadata.getTermsOfUse().getTermsOfUseType()) {
+                        sameLicenseForAllFiles = false;
+                    }
+                    if (firstFileTermsOfUse.getTermsOfUseType() == TermsOfUseType.LICENSE_BASED) {
+                        sameLicenseForAllFiles = (firstFileTermsOfUse.getLicense() != null && fileMetadata.getTermsOfUse().getLicense() != null)  
+                                                 ? firstFileTermsOfUse.getLicense().getId().equals(fileMetadata.getTermsOfUse().getLicense().getId()) : false;
+                    }
+                    if (firstFileTermsOfUse.getTermsOfUseType() == TermsOfUseType.RESTRICTED) {
+                        sameLicenseForAllFiles = firstFileTermsOfUse.getRestrictType() == fileMetadata.getTermsOfUse().getRestrictType() &&
+                                StringUtils.equals(firstFileTermsOfUse.getRestrictCustomText(), fileMetadata.getTermsOfUse().getRestrictCustomText());
+                    }
+                    if (firstFileTermsOfUse.getTermsOfUseType() == TermsOfUseType.ALL_RIGHTS_RESERVED) {
+                        sameLicenseForAllFiles = firstFileTermsOfUse.getRestrictType() == fileMetadata.getTermsOfUse().getRestrictType();
+                    }
+                }
+            }
+
+            if (!dataset.hasActiveEmbargo() && firstFileTermsOfUse != null) {
+                if (sameLicenseForAllFiles) {
+                    if (firstFileTermsOfUse.getTermsOfUseType() == TermsOfUseType.LICENSE_BASED && firstFileTermsOfUse.getLicense() != null) {
+                        solrInputDocument.addField(SearchFields.LICENSE, firstFileTermsOfUse.getLicense().getName());
+                    } else if (firstFileTermsOfUse.getTermsOfUseType() == TermsOfUseType.ALL_RIGHTS_RESERVED) {
+                        solrInputDocument.addField(SearchFields.LICENSE, IndexedTermOfUse.ALL_RIGHTS_RESERVED.getName());
+                    } else if (firstFileTermsOfUse.getTermsOfUseType() == TermsOfUseType.RESTRICTED) {
+                        solrInputDocument.addField(SearchFields.LICENSE, IndexedTermOfUse.RESTRICTED.getName());
```

Comment:
Suggestion: also here we can store `firstFileTermsOfUse.getTermsOfUseType()` and `fileMetadata.getTermsOfUse()` in variables. in order to improve readability.